### ISSUE TITLE
melange: allow to symlink runtime_deps

### ISSUE THREAD

**Issue by jchavarri** [ocaml/dune#7379]
When using melange in a monorepo scenario, where one can have multiple `node_modules` folders across the project, it becomes too expensive to copy these folders into `_build` using `source_tree` or `glob_files_rec`, as Dune has to perform a lot of copy operations, and potentially visit a lot more folders to gather rules (after #7334).

It would be nice to have a way to define a runtime_dep to be copied into the build folder as symlink, e.g.
```
(melange.runtime_deps
 (symlink node_modules))
```
